### PR TITLE
feat: Add events for Notification read / read all

### PR DESCRIPTION
**Pull request by imorland**
**Changes proposed in this pull request:**
Adds events for `Notification` read and readall actions, for extensions to listen for.

**Necessity**

- [ ] Has the problem that is being solved here been clearly explained?
- [ ] If applicable, have various options for solving this problem been considered?
- [ ] For core PRs, does this need to be in core, or could it be in an extension?
- [ ] Are we willing to maintain this for years / potentially forever?

**Confirmed**

- [ ] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `composer test`).
- [ ] Core developer confirmed locally this works as intended.
- [ ] Tests have been added, or are not appropriate here.

**Required changes:**

- [ ] Related documentation PR: (Remove if irrelevant)
- [ ] Related core extension PRs: (Remove if irrelevant)

### CODE REVIEW

**Comment by davwheat** [comment]
Would it be worth also passing the time to the event? There's a chance that any event handlers may wish to use it (for whatever reason), and the time might not be the same as what was set on the DB table.

**Comment by imorland** [comment]
Yeah, I guess that could be useful. Didn't think of that as part of the use case I hace in mind, but usel nonetheless.

File: src/Notification/Command/ReadNotificationHandler.php [hunk 1]
```diff
@@ -10,10 +10,25 @@
...
 
 use Carbon\Carbon;
+use Flarum\Notification\Event\Read;
 use Flarum\Notification\Notification;
+use Illuminate\Contracts\Events\Dispatcher;
 
 class ReadNotificationHandler
 {
+    /**
+     * @var DIspatcher
```
Comment (davwheat) [inline src/Notification/Command/ReadNotificationHandler.php:20]:
```suggestion
     * @var Dispatcher
```

**Review by davwheat — APPROVED** [review]

File: src/Notification/Command/ReadNotificationHandler.php (hunk 1 above)
Comment (imorland) [reply src/Notification/Command/ReadNotificationHandler.php:20]: See, I was just testing to see if anyone noticed 🙈

**Pull request merged**